### PR TITLE
feat(eventing): remove pool events, update deployer and dependencies

### PR DESCRIPTION
**Pull request by datacore-vvarakantham**
Removed pool events as they were added in data plane.

Updated deployer to add eventing to io-engine.

Added nexus events component test to the deployer.

### CODE REVIEW

File: control-plane/agents/src/bin/core/tests/event/mod.rs [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+            ),
+        )
+        .await
+        .expect("Failed to publish volume");
+
+    nexus_creation_event_test(&mut events_mbus_subscription).await;
+
+    let volume_state = volume.state.clone();
+    let nexus = volume_state.target.unwrap();
+    // let nexus_id = NexusId::from(nexus.uuid);
```
Comment (tiagolobocastro) [inline control-plane/agents/src/bin/core/tests/event/mod.rs]:
```suggestion
```

File: control-plane/agents/src/bin/core/tests/event/mod.rs [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+
+    let replicas = api_client.replicas_api().get_replicas().await.unwrap();
+    let testrep = replicas.iter().find(|r| r.node != vol_target).unwrap();
+    let testrep_node = &testrep.node;
+    tracing::info!(
+        "Stopping node {testrep_node} having replica {}",
+        testrep.uri.clone()
```
Comment (tiagolobocastro) [inline control-plane/agents/src/bin/core/tests/event/mod.rs]:
```suggestion
        testrep.uri
```

File: control-plane/grpc/src/operations/volume/traits.rs [hunk 3]
```diff
@@ -1818,29 +1818,25 @@ impl Event for Volume {
...
+            metadata: Some(EventMeta::from_source(event_source)),
         }
     }
 }
 
 // Create volume delete event message.
 impl Event for ValidatedDestroyVolumeRequest {
     fn event(&self) -> EventMessage {
-        let event_meta = EventMeta::new();
+        let event_source = EventSource::new("".to_string());
```
Comment (tiagolobocastro) [inline control-plane/grpc/src/operations/volume/traits.rs]: What is `""` here?

**Review by tiagolobocastro — APPROVED** [review]

File: control-plane/grpc/src/operations/volume/traits.rs (hunk 3 above)
Comment (datacore-vvarakantham) [reply control-plane/grpc/src/operations/volume/traits.rs]: I couldn't get the node name from "ValidatedDestroyVolumeRequest". So I was passing empty string as the node name while creating source.

Comment (tiagolobocastro) [reply control-plane/grpc/src/operations/volume/traits.rs]: Right, I guess depends what we want node_name here to signify..
If we want it to signify the node_name of the running pod perhaps we can, maybe worth a backlog item but for now lgtm.

File: control-plane/agents/src/bin/core/tests/event/mod.rs [hunk 4]
```diff
@@ -0,0 +1,217 @@
...
+    },
+    transport::{DestroyNexus, DestroyPool, DestroyVolume, NexusId, UnpublishVolume, VolumeId},
+};
+use tokio::time::timeout;
+
+const CHILD_WAIT: u64 = 5;
+
+async fn build_cluster(num_ioe: u32, pool_size: u64) -> Cluster {
+    let reconcile_period = Duration::from_millis(300);
+    let child_twait = Duration::from_secs(CHILD_WAIT);
```
Comment (tiagolobocastro) [inline control-plane/agents/src/bin/core/tests/event/mod.rs]: Are we waiting for a full rebuild here? If so let's make this a **lot** smaller than 5s...

File: control-plane/agents/src/bin/core/tests/event/mod.rs [hunk 5]
```diff
@@ -0,0 +1,217 @@
...
+use stor_port::types::v0::{
+    openapi::{
+        models,
+        models::{CreateVolumeBody, PublishVolumeBody, VolumePolicy},
+    },
+    transport::{DestroyNexus, DestroyPool, DestroyVolume, NexusId, UnpublishVolume, VolumeId},
+};
+use tokio::time::timeout;
+
+const CHILD_WAIT: u64 = 5;
```
Comment (tiagolobocastro) [inline control-plane/agents/src/bin/core/tests/event/mod.rs]: Don't make reader guess units by using Duration.

```suggestion
const CHILD_WAIT: Duration = Duration::from_secs(5);
```

**Review by tiagolobocastro — APPROVED** [review]

File: control-plane/grpc/src/operations/volume/traits.rs (hunk 3 above)
Comment (datacore-vvarakantham) [reply control-plane/grpc/src/operations/volume/traits.rs]: Yes. I think it should signify the node where the volume is created. Will add it to TODO.

File: control-plane/agents/src/bin/core/tests/event/mod.rs (hunk 4 above)
Comment (datacore-vvarakantham) [reply control-plane/agents/src/bin/core/tests/event/mod.rs]: Yes, it is full rebuild. Updated it to 1 milli sec.

**Comment by datacore-vvarakantham** [comment]
bors merge

**Comment by tiagolobocastro** [comment]
bors merge

**Comment by datacore-vvarakantham** [comment]
bors merge

**Comment by datacore-vvarakantham** [comment]
bors merge

**Pull request merged**